avocado/core/runner.py

```diff
@@ -373,16 +372,14 @@ def sigtstp_handler(signum, frame):     # pylint: disable=W0613
                     process.kill_process_tree(proc.pid, signal.SIGSTOP, False)
                     self.sigstopped = True
 
-        signal.signal(signal.SIGTSTP, sigtstp_handler)
-
         proc = multiprocessing.Process(target=self._run_test,
                                        args=(test_factory, queue,))
         test_status = TestStatus(self.job, queue)
 
         cycle_timeout = 1
         time_started = time.time()
         proc.start()
-
+        signal.signal(signal.SIGTSTP, sigtstp_handler)
```

Comment:
Yes, it looks more appropriate solution, but don't forget to restore the `SIGTSTP` handler after the test execution, otherwise the next test will already contain this handler.